### PR TITLE
feat(EMI-2159): Auction registration address autocomplete

### PR DESCRIPTION
**Pull request by erikdstock**
The type of this PR is: **FEAT**

This PR solves [EMI-2159] and relates to [EMI-2196]

### Description

This PR adds autocomplete for US addresses on auction registration and makes minor changes to the registration form layout to follow our latest specs for address forms.

It also simplifies the autocomplete input with the tradeoff of eliminating tracking for when a user edits an input on an autocompleted address.

[EMI-2159]: https://artsyproduct.atlassian.net/browse/EMI-2159?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[EMI-2196]: https://artsyproduct.atlassian.net/browse/EMI-2196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Auction/Components/Form/AddressForm.tsx [hunk 1]
```diff
@@ -1,9 +1,42 @@
...
+
+  const trackAutoCompleteEdits = (fieldName: string, handleChange) => (
+    ...args
+  ) => {
+    if (hasAutocompletedAddress) {
+      autocompleteTracking.editedAutocompletedAddress(fieldName)
+      setHasAutocompletedAddress(false)
+    }
+    handleChange(...args)
+  }
```
Comment (damassi) [inline src/Apps/Auction/Components/Form/AddressForm.tsx:39]: Couple things with this:
- if we have a tracking hook that co-locates all tracking, then we shouldn't be defining local functions that also handle tracking; we should figure out how to make it ideal, such as adding a second callback argument to `editedAutocompletedAddress` and then invoking that
- Or (and better) renaming the local function: `handleAutocompleteChange` and then invoking the tracking function from that
- Also, no need to pass in `handleChange` as an argument; its already in scope up above
- Also, we don't need to pass

File: src/Apps/Auction/Components/Form/AddressForm.tsx [hunk 2]
```diff
@@ -80,7 +127,7 @@ export const AddressForm = () => {
         />
       </Column>
 
-      <Column span={6}>
+      <Column span={12}>
```
Comment (damassi) [inline src/Apps/Auction/Components/Form/AddressForm.tsx:130]: Defaults to 12

File: src/Apps/Auction/Components/Form/AddressForm.tsx (hunk 1 above)
Comment (erikdstock) [reply src/Apps/Auction/Components/Form/AddressForm.tsx:39]: yeah, mostly agree here. I'll see if i can improve it a little more.

Comment (erikdstock) [reply src/Apps/Auction/Components/Form/AddressForm.tsx:39]: OK, came back to this and reminded myself how we landed on this structure. The [AddressAutocomplete input exports both its Component and a separate Hook](https://github.com/artsy/force/blob/50611a5316b3b493aff40d96b4e479e180384470/src/Components/Address/AddressAutocompleteInput.tsx#L74) for tracking interactions related to the autocomplete functionality. Normally we would the tracking to exist inside the component, but in this case the job of the autocomplete is to act on the other form inputs - [you can see that down below](https://github.com/artsy/force/pull/14846/files#diff-4517e067c32e835135b2ffc91b55b7d5a0cd01ae8aebe3b1ca2033e7e2c92a38R86-R106). Even so, that change handling could just go in as a prop along with the tracking values like context module up above.

The problem is that we also track when a user _edits_ an autocompleted address and which input that was, which creates a chicken/egg problem between the input and the tracking. We solved it in the first autocomplete implementation (or maybe first iteration on it) by hoisting that one bit of state and tracking functionality up to the same scope as all those inputs in the form of this hook. I remember that another possibility I tried at the time was having the hook return the component itself, like `const { addressAutocompleteTracking, AddressAutocomplete } = useAddressAutocomplete(trackingValues)` so they would still come from a single place, but ultimately this is where we landed. Another option might be a ref or other stateful value.

Finally I wonder if we still need this level of tracking at all. All that said, this is the first time we are bringing the input to a second address form so let me think about best options to move forward.

**Comment by erikdstock** [comment]
After discussing, we'll probably split this into two separate PRs - one for the layout and another for the autocomplete input. I'm going to take a look at what we have so far here and see if it is worth using that or starting from scratch.

**Review by damassi — DISMISSED** [review]

File: src/Apps/Auction/Components/Form/AddressForm.tsx (hunk 2 above)
Comment (erikdstock) [reply src/Apps/Auction/Components/Form/AddressForm.tsx:108]: When I removed this the layout broke so I think the 12 is actually required. Replaced in [4fa5922](https://github.com/artsy/force/pull/14846/commits/4fa5922cd24b93d4bf9b7ffb3b92404b65e2e941).

**Pull request merged**